src/Knp/Snappy/Pdf.php

```diff
@@ -29,19 +32,15 @@ public function __construct($binary = null, array $options = array(), array $env
      */
     protected function handleOptions(array $options = array())
     {
-        $this->hasHtmlHeader = false;
-        $this->hasHtmlFooter = false;
-        
-        if ($this->isFileHeader($options) && !$this->isFile($options['header-html'])) {
-            $options['header-html'] = $this->createTemporaryFile($options['header-html'], 'html');
-        }
-
-        if ($this->isFileFooter($options) && !$this->isFile($options['footer-html'])) {
-            $options['footer-html'] = $this->createTemporaryFile($options['footer-html'], 'html');
-        }
-
-        if ($this->isFileCover($options) && !$this->isFile($options['cover'])) {
-            $options['cover'] = $this->createTemporaryFile($options['cover'], 'html');
+        foreach ($options as $option => $value) {
+            if(null === $value){
```

Comment:
well, it probably can - there are several options without value, like ```toc```which does not have value so now the usage will be ```generate($input, $output, array('toc' => true)``` instead of ```generate($input, $output, array('toc')``` like it was possible in the past. So, it is a possible BC break.